fastlane/actions/project_name.rb

```diff
@@ -0,0 +1,71 @@
+require 'xcodeproj'
+
+module Fastlane
+  module Actions
+    class ProjectNameAction < Action
+
+      PROJECT_EXTENSION = ".xcodeproj".freeze
+
+      # This script returns the application name
+      # if there is a scheme that matches the project name.
+      # Otherwise, it assumes the project name should be specified 
+      # by the user and fails.
+
+      def self.run(params)
+        default_project_name
+      end
+
+      # Fastlane Action class required functions.
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :project_name, optional: true, default_value: default_project_name)
+        ]
+      end
+
+      # Available options default_value helpers
+
+      # In case there is a single '.xcodeproj' in the default directory
+      # it can be automatically inferred by the script 
+      # if no parameter project is received.
+      def self.default_project
+        projects = Dir.entries('.').select { |each| File.extname(each) == PROJECT_EXTENSION }
```

Comment:
Why not use `each.end_with? PROJECT_EXTENSION` as it was before?

Reply:
Because I just found this function in `File` that does what we need, instead of doing it manually. I know a file cannot have more than 1 `.` in the name and `each.end_with? PROJECT_EXTENSION` should also work, but that function is intended to do what we need.